cardano-mempool/tests/Test/Cardano/Memory/PoolTests.hs

```diff
@@ -0,0 +1,174 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module Test.Cardano.Memory.PoolTests (poolTests) where
+
+
+import Cardano.Memory.Pool
+import Common
+import Control.Concurrent (threadDelay)
+import Control.Concurrent.Async
+import Control.Concurrent.Chan
+import Control.Monad
+import Control.Monad.Primitive
+import Data.Bits
+import Data.Function
+import Data.Primitive.PVar
+import Data.Primitive.Ptr (setPtr)
+import Data.Word
+import Foreign.ForeignPtr
+import Foreign.Ptr
+import Foreign.Storable
+import GHC.TypeNats
+import System.Mem (performGC)
+import System.Random.Stateful
+import Test.QuickCheck.Monadic
+
+poolTests :: TestTree
+poolTests = do
+  testGroup
+    "Pool"
+    [ testProperty "findNextZeroIndex" $ propFindNextZeroIndex,
+      poolProps (Block :: Block 32),
+      poolProps (Block :: Block 64)
```

Comment:
We're not going to use this, but might be a good idea to test some non-power-of-2 block sizes.

Would be even better if we could let quickcheck pick them, but I don't think this is possible with the type-safe block sizes.

Reply:
> I don't think this is possible with the type-safe block sizes.

 Sure it is possible :wink: 

> Would be even better if we could let quickcheck pick them

Done